src/intervals/interval_functions.jl

```diff
@@ -1,95 +1,82 @@
 
 ## Powers
-# Integer power of an interval:
+# Integer power:
 function ^{T}(a::Interval{T}, n::Integer)
-    n < zero(n) && return inv( a^(-n) )
-    n == zero(n) && return one(a)
-    n == one(n) && return a
-    #
-    ## NOTE: square(x) is deprecated in favor of x*x
-    if n == 2*one(n)   # this is unnecessary as stands, but  mig(a)*mig(a) is supposed to be more efficient
-        return @round(T, mig(a)^2, mag(a)^2)
-    end
-    #
-    ## even power
-    if n%2 == 0
-        return @round(T, mig(a)^n, mag(a)^n)
-    end
-    ## odd power
-
-    @round(T, a.lo^n, a.hi^n)
+    n < 0  && return inv(a^(-n))
+    n == 0 && return one(a)
+    n == 1 && return a
+
+    iseven(n) && return @round(T, mig(a)^n, mag(a)^n)  # even power
+
+    @round(T, a.lo^n, a.hi^n)  # odd power
 end
 
 ^{T}(a::Interval{T}, r::Rational) = (a^(r.num)) ^ (1/r.den)
 
 # Real power of an interval:
 function ^{T}(a::Interval{T}, x::Real)
-    isinteger(x) && return a^round(Int,x)
-    x < zero(x) && return inv( a^(-x) )
-    x == one(x)/2 && return sqrt(a)
-    #
-    z = zero(a.hi)
-    z > a.hi && error("Undefined operation; Interval is strictly negative and power is not an integer")
-    #
-    xInterv = convert(Interval{T}, x)  # @interval(x) ?
-    diam( xInterv ) >= eps(x) && return a^xInterv
-    # xInterv is a thin interval
-    domainPow = Interval(z, oftype(a.hi, Inf))
-    aRestricted = intersect(a, domainPow)
-    @round(T, aRestricted.lo^x, aRestricted.hi^x)
+    isinteger(x)  && return a^round(Int,x)
+    x < zero(x)  && return inv(a^(-x))
+    x == 0.5  && return sqrt(a)
+
+    zero(a.hi) > a.hi && error("Undefined: interval is strictly negative and power is non-integer")
+
+    x_interval = make_interval(T, x) # convert(Interval{T}, x)  # @interval(x) ?
+    diam(x_interval) > 2*eps(x) && return a^x_interval
```

Comment:
I think the former line was correct, since it is checking if the power is an exact representable number or not, i.e. a thin interval. If it is not, then you promote it into an interval. But if it, then the rest should yield correct results.